pandas/_libs/hashtable_class_helper.pxi.in

```diff
@@ -790,6 +793,9 @@ cdef class StringHashTable(HashTable):
             else:
                 # if ignore_na is False, we also stringify NaN/None/etc.
                 v = get_c_string(<str>val)
+                if v == NULL:
+                    PyErr_Clear()
+                    v = get_c_string(<str>repr(val))
```

Comment:
```suggestion
                    v = get_c_string(repr(val))
```

Shouldn't need this cast

Reply:
If I don't cython, I get the following error:
```
pandas/_libs/hashtable_class_helper.pxi:1495:41: str objects do not support coercion to unicode, use a unicode string literal instead (u'')
```

Reply:
Hmm OK pretty strange; not ideal but I suppose can look in a follow up.

Can you move the location of this code then as requested separately? Want to localize error handling as much as possible

Outside of that lgtm